subprojects/internal-integ-testing/src/test/groovy/org/gradle/integtests/fixtures/logging/TaskGroupingFixtureTest.groovy

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.integtests.fixtures.logging
+
+import spock.lang.Specification
+
+class TaskGroupingFixtureTest extends Specification {
+
+    def "parses task names"() {
+        given:
+        def consoleOutput = """EXECUTING [1s]\u001B[m\u001B[0K\u001B[33D\u001B[1B\u001B[1A\u001B[1m> Task :1:log\u001B[m\u001B[0K
+Output from 1
+
+\u001B[1m> Task :2:log\u001B[m
+Output from 2
+More output from 2
+
+\u001B[1m> Task :3:log\u001B[m
+Output from 3
+
+
+
+Handles lots of newline characters
```

Comment:
Good question.  A task's grouped output always is followed by a newline.  A build status message is always preceded by a newline.  So there should be exactly two newlines between the last task, and the build status, assuming no error.  I will add a test that verifies a task that outputs newlines, that the right number of newlines are present.